dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java

```diff
@@ -319,8 +319,13 @@ protected void githubWebhookDelete(String repository, String gitReference, Strin
         });
 
         // Delete all non-frozen versions that have the same git reference name and then update the file formats of the entry.
+        List<String> entryNamesWithDeletedVersions = new ArrayList<>();
         workflows.forEach(workflow -> {
-            workflow.getWorkflowVersions().removeIf(workflowVersion -> Objects.equals(workflowVersion.getName(), gitReferenceName.get()) && !workflowVersion.isFrozen());
+            Predicate<WorkflowVersion> canVersionBeDeleted = workflowVersion -> Objects.equals(workflowVersion.getName(), gitReferenceName.get()) && !workflowVersion.isFrozen();
+            if (workflow.getWorkflowVersions().stream().anyMatch(canVersionBeDeleted)) {
+                entryNamesWithDeletedVersions.add(computeWorkflowName(workflow));
+            }
+            workflow.getWorkflowVersions().removeIf(canVersionBeDeleted);
             FileFormatHelper.updateEntryLevelFileFormats(workflow);
 
             // Unpublish the workflow if it was published and no longer has any versions
```

Comment:
Looked at that [line](https://github.com/dockstore/dockstore/blob/6dc924e82c0692349a6d6d9026b1af09524cc0ef/dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java#L333) more closely, and the last argument is `false` for the [allowFail](https://github.com/dockstore/dockstore/blob/4ddd48d5bb2c94bf59df95e36afacc0ac9c0df1d/dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubHelper.java#L100) parameter, so it won't throw an exception if it can't find the user.

It seems like we only care if the user exists if the workflow doesn't already exist according to this comment: https://github.com/dockstore/dockstore/blob/6dc924e82c0692349a6d6d9026b1af09524cc0ef/dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java#L544-L545

We also have the same usage here where it won't throw if the user doesn't exist, so it's possible for a `null` user to publish the workflow. (It will throw on this [line](https://github.com/dockstore/dockstore/blob/4ddd48d5bb2c94bf59df95e36afacc0ac9c0df1d/dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java#L521) if the workflow does not exist and the user is null).
https://github.com/dockstore/dockstore/blob/4ddd48d5bb2c94bf59df95e36afacc0ac9c0df1d/dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java#L519

Reply:
Thanks for looking and sorry for the digression; I'd seen that parameter and somehow got the logic backwards and convinced myself it would throw an exception.